Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/Meter/MeterMaintenance.tsx

```diff
@@ -0,0 +1,257 @@
+import { Warning, Modal, LoadingScreen, ServerErrorIcon } from '@gpa-gemstone/react-interactive';
+import { CheckBox, DatePicker } from '@gpa-gemstone/react-forms';
+
+declare var homePath: string;
+
+interface IProps { Meter: OpenXDA.Types.Meter }
+
+//TODO: Move to Gemstone when updating application typings
+interface MaintenanceWindow {
+    ID: number,
+    MeterID: number,
+    StartTime: string,
+    EndTime: string
+}
+
+const MeterMaintenanceWindow = (props: IProps) => {
+    // Table Consts
+    const [sortKey, setSortKey] = React.useState<keyof MaintenanceWindow>('StartTime');
+    const [ascending, setAscending] = React.useState<boolean>(true);
+    const [status, setStatus] = React.useState<Application.Types.Status>('unintiated');
+    const [tableData, setTableData] = React.useState<MaintenanceWindow[]>([]);
+
+    // Modal Consts
+    const [showEditNew, setShowEditNew] = React.useState<boolean>(false);
+    const [showDeleteWarning, setShowDeleteWarning] = React.useState<boolean>(false);
+    const [activeWindow, setActiveWindow] = React.useState<MaintenanceWindow>(null);
+    const [reset, setReset] = React.useState<number>(0);
+
+    React.useEffect(() => {
+        let handle = getMaintenanceWindows();
+        return () => {
+            if (handle != null && handle.abort != null)
+                handle.abort();
+        };
+    }, [ascending, sortKey, props.Meter, reset]);
+
+    function getMaintenanceWindows(): JQuery.jqXHR<MaintenanceWindow[]> {
+        setStatus('loading');
+        let handle = $.ajax({
+            type: 'POST',
+            url: `${homePath}api/OpenXDA/MaintenanceWindow/SearchableList`,
+            contentType: "application/json; charset=utf-8",
+            data: JSON.stringify({
+                Searches: [{ FieldName: 'MeterID', Operator: "=", SearchText: props.Meter.ID, Type: 'string', isPivotColumn: false }],
+                OrderBy: sortKey,
+                Ascending: ascending
+            }),
+            dataType: 'json',
+            cache: false,
+            async: true
+        });
+        handle.done((data) => {
+            setTableData(JSON.parse(data.toString()));
+            setStatus('idle');
+        }).fail(() => setStatus('error'));
+        return handle;
+    }
+
+    function dbMaintenanceWindow(verb: 'POST' | 'DELETE' | 'PATCH', record: MaintenanceWindow): JQuery.jqXHR<MaintenanceWindow[]> {
+        setStatus('loading');
+        let action = (verb === 'POST') ? 'Add' : ((verb === 'DELETE') ? 'Delete' : 'Update');
+        let handle = $.ajax({
+            type: verb,
+            url: `${homePath}api/OpenXDA/MaintenanceWindow/${action}`,
+            contentType: "application/json; charset=utf-8",
+            data: JSON.stringify({ ...record }),
+            dataType: 'json',
+            cache: false,
+            async: true
+        });
+        handle.done((data) => {
+            setReset(reset + 1);
```

Comment:
```suggestion
            setReset(x => x + 1);
```

that'll guarantee an increase otherwise there is potential for a race condition (unlikely but possible)